code/modules/overmap/overmap_object.dm

```diff
@@ -113,8 +127,28 @@ var/global/list/overmap_unknown_ids = list()
 /obj/effect/overmap/proc/adjust_speed(n_x, n_y)
 	CHANGE_SPEED_BY(speed[1], n_x, min_speed)
 	CHANGE_SPEED_BY(speed[2], n_y, min_speed)
+	update_moving()
+
+/obj/effect/overmap/proc/update_moving()
+	if(is_still())
+		if(is_moving)
+			is_moving = FALSE
+			SSovermap.moving_entities -= src
+	else
+		if(!is_moving)
+			is_moving = TRUE
+			SSovermap.moving_entities[src] = TRUE
```

Comment:
This list is deduplicated via `is_moving` already, why assoc?

Reply:
`is_moving` was an artifact of using a processing subsystem, I've changed it to just check list presence now instead.